src/bluetooth/BleService.js

```diff
@@ -52,11 +52,16 @@ class BleService {
    * @param {String} macAddress
    */
   connectAndDiscoverServices = async macAddress => {
-    if (!(await this.manager.isDeviceConnected(macAddress))) {
-      await this.connectToDevice(macAddress);
+    // without the cancel & reconnect further commands
+    // were sometimes returning an error: "BleError: Device [mac address] was disconnected"
+    if (await this.manager.isDeviceConnected(macAddress)) {
+      await this.manager.cancelDeviceConnection(macAddress);
     }
 
-    return this.manager.discoverAllServicesAndCharacteristicsForDevice(macAddress);
+    const device = await this.connectToDevice(macAddress, { autoConnect: true });
```

Comment:
Hm, does the `autoConnect` arg get passed through to the BleManager if `connectToDevice` doesn't also pass  through? 🤔  https://github.com/openmsupply/mobile/blob/1f2149f1c6861883579072786747316b4fc9915e/src/bluetooth/BleService.js#L44

Reply:
DOH! good spotting @katherine-sussol